Fix postgresql_lang integration test for CentOS 8

samdoran opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
`test/integration/targets/postgresql_lang/`

samdoran committed:
Fix postgresql_lang integration test for CentOS 8

Andersson007 reviewed (commented):
@samdoran 
I don't know why pg_dir and pg_hba_location variables aren't visible in test/integration/targets/setup_postgresql_db/tasks/main.yml anymore (with changes in this PR)
If i change the following parts there it works OK
**1.**
```
 68 - name: Initialize postgres (RedHat systemd)
 69   command: postgresql-setup initdb
 70   when: ansible_os_family == "RedHat" and ansible_service_mgr == "systemd"
```
to
```
 68 - name: Initialize postgres (RedHat systemd)
 69   become: yes
 70   become_user: "{{ pg_user }}"
 71   shell: initdb -D /var/lib/pgsql/data
 72   when: ansible_os_family == "RedHat" and ansible_service_mgr == "systemd"
```
**2.** and
```
 86 - name: Copy pg_hba into place
 87   template:
 88     src: files/pg_hba.conf
 89     dest: "{{ pg_hba_location }}"
 90     owner: "{{ pg_user }}"
 91     group: "{{ pg_group }}"
 92     mode: "0644"
```
to
```
 88 - name: Copy pg_hba into place
 89   template:
 90     src: files/pg_hba.conf
 91     dest: /var/lib/pgsql/data
 92     owner: "{{ pg_user }}"
 93     group: "{{ pg_group }}"
 94     mode: "0644"
```
I changed the first block completely using a path instead of variable and another command to initialize it.
In the second part i only changed the variable to path

samdoran committed:
Only search inside the role for files to include

When a role is called as a dependency, both roles are searched. This can cause the incorrect file to be included if it is picked up from the parent role.

samdoran commented:
That was due to a rather interesting behavior I spent some time tracking down today. When a role is called as a dependency, the role path of the calling role is used when searching paths for the `first_found` look. Since I used the same file name in `postgresql_lang/vars` as in `setup_postgresql_db/vars`, the lookup found the file in `postgresql_lang/vars`, which did not contain the correct values.

To fix this, I set the search path to an absolute path to the dependent role: `{{ role_path }}/vars`.

That had me confused for a quite a while. 😕

Andersson007 reviewed (approved):
@samdoran thanks for fixing this!
shipit
